File: pirates-ios/PiratesGame/Page/Modules/ControllerModule.swift
//
//  ControllerModule.swift
//  Page module: provides a protocol for separating controller logic into modules
//

import UIKit
import JsonInflator

protocol ControllerModule: class {
    
    var eventType: String { get }
    
    func didCreate(viewController: UIViewController)
    func didUpdatePage(page: Page, binder: InflatorDictBinder)
    func catchEvent(_ event: AppEvent, sender: Any?) -> Bool

}

File: pirates-android/app/src/main/java/com/crescentflare/piratesgame/page/modules/ControllerModule.kt
package com.crescentflare.piratesgame.page.modules

import android.content.Context
import com.crescentflare.jsoninflator.binder.InflatorMapBinder
import com.crescentflare.piratesgame.infrastructure.events.AppEvent
import com.crescentflare.piratesgame.page.storage.Page

/**
 * Page module: provides an interface for separating controller logic into modules
 */
interface ControllerModule {

    val eventType: String

    fun onCreate(context: Context)
    fun onPageUpdated(page: Page, binder: InflatorMapBinder)
    fun catchEvent(event: AppEvent, sender: Any?): Boolean

}
